ui/src/importer/importerTaskDetails/cancelImportTask.js

```diff
@@ -0,0 +1,89 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Button, Confirm, Message } from 'semantic-ui-react';
+import { importerApi } from '../../api/importer';
+
+export class CancelImportTask extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      modalOpen: false,
+      cancelFailed: false,
+      cancelLoading: false,
+      taskCancelled: undefined,
+    };
+  }
+
+  handleCancelImportAction = async () => {
+    const { logId } = this.props;
+    try {
+      this.setState({ cancelLoading: true });
+      const response = await importerApi.cancel(logId);
```

Comment:
You should use `withCancel` pattern here to avoid memory leaks, given that you are not using Redux.